litmus-portal/frontend/src/pages/MonitoringDashboardPage/index.tsx

```diff
@@ -235,17 +754,200 @@ const DashboardPage: React.FC = () => {
+                  )}`}
+
+                  <IconButton className={classes.rangeSelectorIcon}>
+                    <KeyboardArrowDownIcon />
+                  </IconButton>
+                </Typography>
+              </ButtonOutlined>
+              <DateRangeSelector
+                anchorEl={dateRangeSelectorRef.current as HTMLElement}
+                isOpen={isDateRangeSelectorPopoverOpen}
+                onClose={() => {
+                  setDateRangeSelectorPopoverOpen(false);
+                }}
+                callbackToSetRange={CallbackFromRangeSelector}
+                className={classes.rangeSelectorPopover}
+              />
+
+              <FormControl className={classes.formControl} variant="outlined">
+                <InputLabel
+                  id="refresh-controlled-open-select-label"
+                  className={classes.inputLabel}
+                >
+                  <AutorenewOutlinedIcon className={classes.refreshIcon} />
+                  <Typography className={classes.refreshText}>
+                    Refresh
+                  </Typography>
+                </InputLabel>
+                <Select
+                  labelId="refresh-controlled-open-select-label"
+                  id="refresh-controlled-open-select"
+                  open={openRefresh}
+                  disabled={getRefreshRateStatus()}
+                  onClose={handleCloseRefresh}
+                  onOpen={handleOpenRefresh}
+                  value={refreshRate !== 0 ? refreshRate : null}
+                  onChange={(event: React.ChangeEvent<{ value: unknown }>) => {
+                    // When viewing data for non-relative time range, refresh should be Off ideally.
+                    // UI can auto detect if it is not Off and switches it to Off.
+                    // Now the user can try to view the non-relative time range data again.
+                    if (selectedDashboard.refreshRate !== 2147483647) {
+                      dashboard.selectDashboard({
+                        refreshRate: event.target.value as number,
+                      });
+                      setRefreshRate(event.target.value as number);
+                    } else {
+                      dashboard.selectDashboard({
+                        refreshRate: event.target.value as number,
+                      });
+                      setRefreshRate(event.target.value as number);
+                      dashboard.selectDashboard({
+                        forceUpdate: true,
+                      });
+                      setPrometheusQueryData({
+                        ...prometheusQueryData,
+                        firstLoad: true,
+                      });
+                    }
+                  }}
+                  input={<OutlinedInput classes={outlinedInputClasses} />}
+                  IconComponent={KeyboardArrowDownIcon}
+                >
+                  <MenuItem
+                    key="Off-refresh-option"
```

Comment:
key is not required here

Reply:
I am getting some linting errors without that.